core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -75,6 +75,17 @@ private[spark] class HadoopDelegationTokenManager(
       .toMap
   }
 
+  private def safeCreateProvider(
+      createFn: => HadoopDelegationTokenProvider): Option[HadoopDelegationTokenProvider] = {
+    try {
+      Some(createFn)
+    } catch {
+      case t: Throwable =>
+        logDebug(s"Failed to load built in provider.", t)
```

Comment:
perhaps this should be a warn?

Reply:
info probable - it is actually ok for this to fail if provider is not relevant (and in classpath).

Reply:
I think debug is right, actually -- we have no idea at this point if the user wants these credential providers, and it could be totally fine if they're missing eg. if they never want to talk to hive.

(also don't really care that much and don't want to bike-shed on this ...)

Reply:
I actually think this really should be debug. This only covers exceptions thrown by the constructor, which really shouldn't be doing anything (and this code is just needed because of the class linkage issue).

Individual methods like `obtainDelegationTokens` should be the ones reporting user-actionable issues, and even them today kinda log most things at debug level...